tests/unit/type/list-3.tcl

```diff
@@ -47,39 +100,42 @@ foreach comp {2 1 0} {
     
     test "Stress tester for #3343-alike bugs comp: $comp" {
         r del key
+        set sent {}
+        set print_commands false
         for {set j 0} {$j < $cycles} {incr j} {
-            set op [randomInt 7]
-            set small_signed_count [expr 5-[randomInt 10]]
-            if {[randomInt 2] == 0} {
-                set ele [randomInt 1000]
-            } else {
-                set ele [string repeat x [randomInt 10000]][randomInt 1000]
-            }
-            switch $op {
-                0 {r lpush key $ele}
-                1 {r rpush key $ele}
-                2 {r lpop key}
-                3 {r rpop key}
-                4 {
-                    catch {r lset key $small_signed_count $ele}
-                }
-                5 {
-                    set otherele [randomInt 1000]
-                    if {[randomInt 2] == 0} {
-                        set where before
-                    } else {
-                        set where after
-                    }
-                    r linsert key $where $otherele $ele
-                }
-                6 {
-                    set index [randomInt [r llen key]]
-                    set otherele [r lindex key $index]
-                    r lrem key 1 $otherele
+            if {[catch {
+                set cmd [generate_cmd_on_list_key key]
+                lappend sent $cmd
+
+                # execute the command, we expect commands to fail on syntax errors
+                r {*}$cmd
+            }]} {
+                if {[count_log_message 0 "crashed by signal"] != 0 || [count_log_message 0 "ASSERTION FAILED"] != 0} {
+                    puts "Server crashed"
+                    set print_commands true
+                    break
                 }
             }
         }
-    }
+
+        # check valgrind and asan report for invalid reads after execute
+        # command so that we have a report that is easier to reproduce
+        set valgrind_errors [find_valgrind_errors [srv 0 stderr] false]
+        set asan_errors [sanitizer_errors_from_file [srv 0 stderr]]
+        if {$valgrind_errors != "" || $asan_errors != ""} {
+            puts "valgrind or asan found an issue"
+            set print_commands true
+        }
+
+        if {$print_commands} {
+            puts "violating commands:"
+            foreach cmd $sent {
+                puts $cmd
+            }
+        }
+
+        r ping
+    } {PONG} {external:skip}
```

Comment:
not sure i like to skip it on external tests.
maybe we can just skip the valgrind / asan check in external mode?